### ISSUE TITLE
Feature Request: Store Connection Information in Service Connection

### ISSUE THREAD

**Issue by ChristophHannappel** [thycotic/TSS-ADO-BuildTask#3]
Hi,
i've got a feature request. We're using Azure DevOps Server and don't have an Azure Key Vault. So in order to use the Thycotic TSS Task i've to put Credentials inside the Build Yaml, which is obviously a bad idea. Also it defeats the use of the TSS, if i have to setup auzre key vault.
Could you add support for a "Service Conenction" where i can store the Secret Server URL, API Path Uri, Token Path Uri, Username and Password. Also make the Task configurable to use the service connection.

This also reduces the effort for configuring each task.

Thank you very much.

**Comment by nichols-green** [comment]
This really should have been the design from the start. Until this is implemented, you should be able to add the sensitive connection information as a variable in the library tab of the pipelines section. 

https://docs.microsoft.com/en-us/azure/devops/pipelines/library/variable-groups?view=azure-devops&tabs=yaml 

Just be sure to set the variable type to secret by clicking the lock on each variable.

**Comment by pvanulden** [comment]
As @nichols-green has indicated, in the absence of Key Vault, the best way to store credentials is via variable groups in your pipeline library which you can "lock" after entering them. While it would certainly be convenient to store everything within a Service Connection, that does goes against principles of least privilege. If you have to access multiple secrets within your pipeline, they should really have their own individual credentials anyway.

**Comment by nichols-green** [comment]
I'm willing to accept that this is the direction the Thycotic team is going with, but I also think supporting service connections is a standardized and widely adopted pattern in Azure DevOps. Not only is it convenient, the scoped access controls allow a team to define a smaller subset of team members access to setup and modify connections, while granting the entire team the ability to consume defined connections. Managing these connection details in variable groups isn't difficult, but it's not the expected standard practice here either.

**Comment by golaat** [comment]
I agree with @nichols-green. Service Connections in Azure DevOps are the way to go. They provide a centralized place to store and managed access to external systems. They also provide an additional approval mechanism for their assignment and use in pipelines.
